regulations/views/comment.py

```diff
@@ -70,62 +66,66 @@ def preview_comment(request):
     sections = body.get('assembled_comment', [])
     html = tasks.json_to_html(sections)
     key = '/'.join([settings.ATTACHMENT_PREVIEW_PREFIX, get_random_string(50)])
-    s3 = tasks.make_s3_client()
     with tasks.html_to_pdf(html) as pdf:
-        s3.put_object(
+        tasks.s3_client.put_object(
             Body=pdf,
             ContentType='application/pdf',
             ContentDisposition='attachment; filename="comment.pdf"',
             Bucket=settings.ATTACHMENT_BUCKET,
             Key=key,
         )
-    url = s3.generate_presigned_url(
-        ClientMethod='get_object',
-        Params={
-            'Bucket': settings.ATTACHMENT_BUCKET,
-            'Key': key,
-        },
-    )
+    _, url = tasks.SignedUrl.generate(key=key)
     return JsonResponse({'url': url})
 
 
-@csrf_exempt
-@require_http_methods(['POST'])
-def submit_comment(request):
-    """Submit a comment to the task queue. The request body is JSON
-    with an 'assembled_comment' field and additional fields.
-    """
-    body = json.loads(request.body.decode('utf-8'))
-    valid, message = docket.sanitize_fields(body)
-    if not valid:
-        logger.error(message)
-        return JsonResponse({'message': message}, status=403)
+class SubmitCommentView(View):
 
-    files = tasks.extract_files(body['assembled_comment'])
-    # Account for the main comment itself submitted as an attachment
-    if len(files) > settings.MAX_ATTACHMENT_COUNT - 1:
-        message = "Too many attachments"
-        logger.error(message)
-        return JsonResponse({'message': message}, status=403)
-
-    s3 = tasks.make_s3_client()
-    metadata_key = get_random_string(50)
-    metadata_url = s3.generate_presigned_url(
-        ClientMethod='get_object',
-        Params={
-            'Bucket': settings.ATTACHMENT_BUCKET,
-            'Key': metadata_key,
-        },
-    )
-    chain = celery.chain(
-        tasks.submit_comment.s(body),
-        tasks.publish_tracking_number.s(key=metadata_key),
-    )
-    chain.delay()
-    return JsonResponse({
-        'status': 'submitted',
-        'metadata_url': metadata_url,
-    })
+    def post(self, request, doc_number):
+        form = {
+            key: value for key, value in request.POST.items()
+            if key not in {'comments', 'csrfmiddlewaretoken'}
```

Comment:
I understand the desire to exclude `comments`, but let's avoid spelling out `csrfmiddlewaretoken` -- it'll be stripped in the sanitization step (as it would be if we used any other form framework). My concern is around readability.